src/test/java/core/basesyntax/serviceimpl/TransactionServiceImplTest.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.TransactionService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class TransactionServiceImplTest {
+    private TransactionService transactionService;
+    private FruitStrategy fruitStrategy;
+
+    @BeforeEach
+    void setUp() {
+        fruitStrategy = new FruitStrategy(Map.of(
+                FruitTransaction.Operation.BALANCE,new BalanceOperationHandler(),
+                FruitTransaction.Operation.SUPPLY,new SupplyReturnOperationHandler(),
+                FruitTransaction.Operation.PURCHASE,new PurchaseOperationHandler()));
+        transactionService = new TransactionServiceImpl(fruitStrategy);
+    }
+
+    @Test
+    void processData_Ok() {
+        List<FruitTransaction> list = new ArrayList<>();
+        list.add(new FruitTransaction(FruitTransaction.Operation.BALANCE,"apple",20));
+        list.add(new FruitTransaction(FruitTransaction.Operation.SUPPLY,"apple",5));
+        list.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE,"apple",10));
+        transactionService.processData(list);
+        Assertions.assertEquals(15,Storage.DB.get("apple"));
+    }
+
+    @Test
+    void fruitTransaction_OK() {
```

Comment:
fruitTransaction_OK?
Do we test `fruitTransaction `method?

Reply:
I wanted to cover all lines. because  i had error in fruitTransaction class, that expected covered lines must be more 0.8 but was 0.6.

Reply:
You can do it - just choose the correct name for test methods. For example
processData_oneTransaction_Ok
processData_manyTransactions_Ok
processData_onlyBalanceTransactions_Ok

Reply:
thanks. Now i understand.